misc/download-redis

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+# This script downloads the contents of the cache from a Redis secondary storage
+
+import redis
+import os
+
+import progress.bar
+import progress.spinner
+import humanize
+
+config = os.getenv("REDIS_CONF", "localhost")
+if ":" in config:
+    host, port = config.rsplit(":", 1)
+    sock = None
+elif config.startswith("/"):
+    host, port, sock = None, None, config
+else:
+    host, port, sock = config, 6379, None
+username = os.getenv("REDIS_USERNAME")
+password = os.getenv("REDIS_PASSWORD")
+context = redis.Redis(host=host, port=port, unix_socket_path=sock, password=password)
+pipe = context.pipeline(transaction=False)
+
+ccache = os.getenv("CCACHE_DIR", os.path.expanduser("~/.cache/ccache"))
+try:
+    count = context.info()["db0"]["keys"]
+except:
+    count = None  # only for showing progress
+files = result = manifest = objects = 0
+size = 0
+columns = os.get_terminal_size()[0]
+width = min(columns - 24, 100)
+bar = progress.bar.Bar(
+    "Downloading...", max=count, fill="=", suffix="%(percent).1f%%", width=width
+)
+if not count:
+    bar = progress.spinner.Spinner("Downloading... ")
+keys = context.scan_iter()
```

Comment:
Perhaps use

```suggestion
keys = context.scan_iter(match="ccache:*")
```

here? Then there will be no need to filter out keys not starting with `ccache:` later.

Reply:
It was done that way mostly for the progress bar to match, but also some servers didn't implement "match"